flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/FieldAccessFromTableITCase.java

```diff
@@ -0,0 +1,111 @@
+import static org.apache.flink.table.api.DataTypes.STRING;
+import static org.apache.flink.table.api.Expressions.$;
+
+/**
+ * Regular tests. See also {@link ConstructedAccessFunctionsITCase} for tests that access a nested
+ * field of an expression or for {@link BuiltInFunctionDefinitions#FLATTEN} which produces multiple
+ * columns from a single one.
+ */
+class FieldAccessFromTableITCase extends BuiltInFunctionTestBase {
+
+    @Override
+    public Stream<TestSetSpec> getTestCaseSpecs() {
+        return Stream.of(
+
+                // Actually in case of SQL it does not use the GET method, but
+                // a custom logic for accessing nested fields of a Table.
+                TestSetSpec.forFunction(BuiltInFunctionDefinitions.GET)
+                        .onFieldsWithData(null, Row.of(1))
+                        .andDataTypes(
+                                ROW(FIELD("nested", BIGINT().notNull())).nullable(),
+                                ROW(FIELD("nested", BIGINT().notNull())).notNull())
+                        .testResult(
+                                resultSpec(
+                                        $("f0").get("nested"),
+                                        "f0.nested",
+                                        null,
+                                        BIGINT().nullable()),
+                                resultSpec(
+                                        $("f1").get("nested"),
+                                        "f1.nested",
+                                        1L,
+                                        BIGINT().notNull())),
+
+                // In Calcite it maps to FlinkSqlOperatorTable.ITEM
+                TestSetSpec.forFunction(BuiltInFunctionDefinitions.AT)
+                        .onFieldsWithData(
+                                null,
+                                new int[] {1},
+                                null,
+                                singletonMap("nested", 1),
+                                null,
+                                Row.of(1))
+                        .andDataTypes(
+                                ARRAY(BIGINT().notNull()).nullable(),
+                                ARRAY(BIGINT().notNull()).notNull(),
+                                MAP(STRING(), BIGINT().notNull()).nullable(),
+                                MAP(STRING(), BIGINT().notNull()).notNull(),
+                                ROW(FIELD("nested", BIGINT().notNull())).nullable(),
+                                ROW(FIELD("nested", BIGINT().notNull())).notNull())
+                        // accessing elements of MAP or ARRAY is a runtime operations,
+                        // we do not know about the size or contents during the inference
+                        // therefore the results are always nullable
+                        .testResult(
+                                resultSpec($("f0").at(1), "f0[1]", null, BIGINT().nullable()),
```

Comment:
remove the outer `resultSpec(...)`, `testResult` can take it directly. also in other tests.

Reply:
in this case i'm not sure it does. `testResult` is either a vararg of result spec or a function taking 5 args. I rather prefer to keep it as it is.

Reply:
`BuiltInFunctionTestBase.TestSpec#testResult(org.apache.flink.table.expressions.Expression, java.lang.String, java.lang.Object, org.apache.flink.table.types.AbstractDataType<?>)` has 4 args as well.